vimeo-networking/src/main/java/com/vimeo/networking/VimeoClient.java

```diff
@@ -922,10 +953,13 @@ public void failure(VimeoError error) {
     // -----------------------------------------------------------------------------------------------------
     // <editor-fold desc="Editing (Video, User)">
     @Nullable
-    public Call<Object> editVideo(String uri, String title, String description, @Nullable String password,
-                                  @Nullable Privacy.PrivacyValue privacyValue,
-                                  @Nullable HashMap<String, Object> parameters,
-                                  ModelCallback<Object> callback) {
+    public Call<Video> editVideo(String uri,
+                                 String title,
+                                 String description,
+                                 @Nullable String password,
+                                 @Nullable Privacy.PrivacyValue privacyValue,
+                                 @Nullable HashMap<String, Object> parameters,
+                                 VimeoCallback<Video> callback) {
```

Comment:
If we are throwing exceptions when things like `uri` or `callback` are null, why not just enforce `NotNull` on those parameters and remove the checks? I feel like this is a nice juncture to do so. This applies to most methods in this class. I don't think it's a ton of work to do, but I know you have already spent a lot of time on this, so I leave it up to your discretion. If you don't want to tackle it now, perhaps add a ticket so we can revisit.